src/CMakeLists.txt

```diff
@@ -10,17 +10,16 @@ set(UMF_SOURCES
     memory_pool.c
     memory_provider.c
     memory_provider_get_last_failed.c
-    provider/provider_os_memory.c
```

Comment:
please do not change any windows - related code here

Reply:
Why? I don't see any reason for keeping files that just have stubs. 

Also, this PR makes it necessary to store extra state (hwloc_topology) in OS provider struct. It's easier to keep it in an OS-agnostic way vs defining a struct that is OS-specific. We will probably move the hwloc_topology to another place anyway in future (to some library ctor or `initialize` method so we can support discovery).

Reply:
ok let's get back to this when we start adding support for windows